lib/puma/request.rb

```diff
@@ -225,12 +235,13 @@ def prepare_response(status, headers, res_body, requests, client)
       io_buffer << line_ending
 
       if response_hijack
-        fast_write_str socket, io_buffer.to_s
+        fast_write_str socket, io_buffer.rd_and_rst
```

Comment:
```suggestion
        fast_write_str socket, io_buffer.read_and_reset
```

Reply:
Fixed, pushed back into 'Updates for response body processing'